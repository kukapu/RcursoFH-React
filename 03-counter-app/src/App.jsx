import PropTypes from 'prop-types';


export const App = ( { 
    title,
    subTitle = 'El porro vengador',
    name 
} ) => {
 
    // console.log(title)

    return (
        <>
            <h1 data-testid='test-title'>{ title }</h1>
            {/* <code>{ JSON.stringify(newMessage) }</code> */}
            <p>{ subTitle }</p>
            <p>{ subTitle }</p>
            <p>{ subTitle }</p>
            <p>{ name }</p>
        </>
    )
}

App.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}

App.defaultProps = {
    name: 'Porros escondidos',
    subTitle: 'No hay subtitulo',
    // title: 'No hay titulo',
}