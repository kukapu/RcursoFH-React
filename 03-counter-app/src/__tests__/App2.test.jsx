import { render, screen } from '@testing-library/react';
import { App } from '../App';

describe('Pruebas en <App />ยบ', () => { 
    
    const title = 'Hola, Soy Goku'
    const subTitle = 'Soy un subtitulo'

    test('should match with snapshot', () => { 
        
        const { container } = render( <App title={ title } /> )
        expect( container ).toMatchSnapshot()

    })

    test('should show message "Hola, Soy Goku"', () => { 

        render( <App title={ title } /> )
        // screen.debug()
        expect( screen.getByText(title) ).toBeTruthy()

    })

    test('should show title in h1', () => {
        
        render( <App title={ title } /> )
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title )


    })

    test('should show subTitle send for props', () => {

        render(
             <App 
                title={ title }
                subTitle={ subTitle }
            />
        )

        expect( screen.getAllByText(subTitle).length ).toBe(3)
        
    })


})