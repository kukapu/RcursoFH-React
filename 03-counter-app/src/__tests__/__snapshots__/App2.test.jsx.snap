// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App />ยบ should match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Goku
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Porros escondidos
  </p>
</div>
`;
